Documentation/ApiOverview/DependencyInjection/Index.rst

```diff
@@ -714,6 +714,36 @@ be declared public:
     Call to a member function methodName() on null
 
 
+How to override service arguments?
+----------------------------------
+
+Some services in the TYPO3 core use service arguments, which can be overridden by third-party extensions.  
+For example, the :php:`$rateLimiterFactory` argument in the :php:`PasswordRecoveryController` of the 
+:php:`felogin` extension  uses a service with the ID :php:`feloginPasswordRecovery.rateLimiterFactory`. 
+This service is defined in the :php:`Services.yaml` file of :php:`ext:felogin` and includes a service 
+argument named :php:`$config`, which specifies the configuration for the Symfony Rate Limiter used in 
+the class.
+
+A third-party extension can override the :php:`feloginPasswordRecovery.rateLimiterFactory` service in 
+its own :php:`Services.yaml` file, as shown in the example below:
+
+.. code-block:: php
+
+    feloginPasswordRecovery.rateLimiterFactory:
+      class: Symfony\Component\RateLimiter\RateLimiterFactory
+      arguments:
+        $config:
+          id: 'felogin-password-recovery'
+          policy: 'sliding_window'
+          limit: 10
+          interval: '30 minutes'
+        $storage: '@storage.cachingFramework'
+
+It is important that the 3rd party extension is loaded **after** the extension whose service it overrides.
+This can be achieved by requiring the original extension as a dependency in the :php:`composer.json` file
```

Comment:
Do we not have a central place where extension ordering is explained and link to it before we leave that here only quarter-baked ?

* composer.json: require or suggests section (composer mode)
* ext_emconf.php: depends or depends section (classic mode)

Reply:
We could link to https://docs.typo3.org/permalink/t3coreapi:confval-ext-emconf-constraints (classic). We also write about it in https://docs.typo3.org/permalink/t3coreapi:ext-composer-json-minimal but maybe not as extended as we should.

Reply:
Good point. I suggest we should have a dedicated section under https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ExtensionArchitecture/BestPractises/Index.html about extension load order? Is this a way to go? If yes, I can take care of writing the docs for that section.

Reply:
That would be great! and then I would link it from ext-emconf and composer json links.

Reply:
@derhansen please open an issue or PR about this